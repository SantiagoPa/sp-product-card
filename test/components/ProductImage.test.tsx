import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components/';
import { product2 } from '../data/products';

describe('Porduct Title', () => {
    
  test('debe de mostrar el componente correctamente con el url de la imagen personalizada', () => {
    const wrapper = renderer.create(<ProductImage img="www.imagen.com" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componente con la url del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        { () => <ProductImage img={product2.img} /> }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
