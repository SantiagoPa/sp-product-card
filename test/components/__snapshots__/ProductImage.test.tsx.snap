// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Porduct Title debe de mostrar el componente con la url del producto 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Porduct Title debe de mostrar el componente correctamente con el url de la imagen personalizada 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="www.imagen.com"
/>
`;
